sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/KeyResolver.cs

```diff
@@ -0,0 +1,184 @@
+        /// <returns>A new <see cref="CryptographyClient"/> capable of performing cryptographic operations with the key represented by the specfiied keyId</returns>
+        public virtual CryptographyClient Resolve(Uri keyId, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(keyId, nameof(keyId));
+
+            using DiagnosticScope scope = _pipeline.Diagnostics.CreateScope("Azure.Security.KeyVault.Keys.Cryptography.KeyResolver.Resolve");
+            scope.AddAttribute("key", keyId);
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNull(keyId, nameof(keyId));
+
+                Key key = GetKey(keyId, cancellationToken);
+
+                KeyVaultPipeline pipeline = new KeyVaultPipeline(keyId, _apiVersion, _pipeline);
+
+                return (key != null) ? new CryptographyClient(key.KeyMaterial, pipeline) : new CryptographyClient(keyId, pipeline);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a <see cref="CryptographyClient"/> capable of performing cryptographic operations with the key represented by the specfiied keyId.
+        /// </summary>
+        /// <param name="keyId">The key idenitifier of the key used by the created <see cref="CryptographyClient"/> </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A new <see cref="CryptographyClient"/> capable of performing cryptographic operations with the key represented by the specfiied keyId</returns>
+        public virtual async Task<CryptographyClient> ResolveAsync(Uri keyId, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(keyId, nameof(keyId));
+
+            using DiagnosticScope scope = _pipeline.Diagnostics.CreateScope("Azure.Security.KeyVault.Keys.Cryptography.KeyResolver.Resolve");
+            scope.AddAttribute("key", keyId);
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNull(keyId, nameof(keyId));
+
+                Key key = await GetKeyAsync(keyId, cancellationToken).ConfigureAwait(false);
+
+                KeyVaultPipeline pipeline = new KeyVaultPipeline(keyId, _apiVersion, _pipeline);
+
+                return (key != null) ? new CryptographyClient(key.KeyMaterial, pipeline) : new CryptographyClient(keyId, pipeline);
+
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        IKeyEncryptionKey IKeyEncryptionKeyResolver.Resolve(Uri keyId, CancellationToken cancellationToken)
+        {
+            return ((KeyResolver)this).Resolve(keyId, cancellationToken);
```

Comment:
`((KeyResolve)this)` shouldn't be necessary (VS should prompt to "simplify"). Code consistency.